src/conversion/onnx_to_krnl/tensor/padconstantvaluepad.cpp

```diff
@@ -0,0 +1,108 @@
+//===----padconstantvaluepad.cpp - Lowering PadConstantValuePad Op --------===//
+//
+// Copyright 2019 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX PadConstantValuePad  Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/conversion/onnx_to_krnl/onnx_to_krnl_common.hpp"
+
+using namespace mlir;
+
+struct ONNXPadConstantValuePadOpLowering : public ConversionPattern {
+  ONNXPadConstantValuePadOpLowering(MLIRContext *ctx)
+      : ConversionPattern(mlir::ONNXPadConstantValuePadOp::getOperationName(),
+                          1, ctx) {}
+
+  PatternMatchResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    auto tensorType = (*op->result_type_begin()).cast<TensorType>();
```

Comment:
Since result type can be MemRefType in op definition, this casting to TensorType is potentially failed. It is enough to use `auto tensorType = *op->result_type_begin()` since the following `convertToMemRefType` supports both MemRefType and TensorType.

Reply:
I see this change hasn't been made yet so I will postpone merging this. @chentong319 could you have a look at this?

Reply:
@doru1004: @tungld pointed out an error in interpreting pad value. Please do not merge until I fix it.

Reply:
@chentong319 sure no problem, I'll wait for you to let me know when the patch is ready to land.